File: Backend Code/src/main/java/com/LabResourceUtilizationPlatform/Service/EquipmentService.java
package com.LabResourceUtilizationPlatform.Service;

import com.LabResourceUtilizationPlatform.Dtos.Request.CreateEquipmentRequest;
import com.LabResourceUtilizationPlatform.Dtos.Request.UpdateEquipmentRequest;
import com.LabResourceUtilizationPlatform.Dtos.Response.EquipmentDetailResponse;
import com.LabResourceUtilizationPlatform.Dtos.Response.EquipmentResponse;
import com.LabResourceUtilizationPlatform.Entity.Enum.EquipmentStatus;
import org.springframework.web.multipart.MultipartFile;


import java.util.List;
import java.util.Map;

public interface EquipmentService {

    EquipmentResponse createEquipment(CreateEquipmentRequest request, MultipartFile image);

    EquipmentResponse getEquipmentByCode(
            String equipmentCode,
            String labCode,
            String institutionCode);

    List<EquipmentResponse> getAllEquipment(
            String labCode,
            String institutionCode);

    EquipmentResponse updateEquipment(UpdateEquipmentRequest request);

    void deleteEquipment(
            String equipmentCode,
            String labCode,
            String institutionCode);

    Map<String, Long> getEquipmentStatusCounts(String institutionCode);

    EquipmentDetailResponse getEquipmentDetail(
            String equipmentCode,
            String labCode,
            String institutionCode);

    List<EquipmentResponse> getDepartmentEquipment(String institutionCode, String departmentName);

    void updateStatus(String equipmentCode, EquipmentStatus status);
}

File: Frontend Code/src/services/equipmentService.js
import api from "./api";

const equipmentService = {
  create: async (data) => {
    const response = await api.post("/equipment", data, {
      headers: {
        "Content-Type": "multipart/form-data",
      },
    });

    return response.data;
  },

  getDepartmentEquipment: async (institutionCode, departmentName) => {
    const response = await api.get("/equipment/department", {
      params: {
        institutionCode,
        departmentName,
      },
    });
    return response.data;
  },

  getByCode: async (institutionCode, labCode, equipmentCode) => {
    const response = await api.get(
      `/equipment/${institutionCode}/${labCode}/${equipmentCode}`
    );
    return response.data;
  },

  getByLab: async (institutionCode, labCode) => {
    const response = await api.get(
      `/equipment/${institutionCode}/${labCode}`
    );
    return response.data;
  },

  updateStatus: async (equipmentCode, status) => {
    const response = await api.patch(
      `/equipment/${equipmentCode}/status`,
      null,
      {
        params: { status },
      }
    );

    return response.data;
  },

  update: async (data) => {
    const response = await api.put("/equipment", data);
    return response.data;
  },

  delete: async (institutionCode, labCode, equipmentCode) => {
    const response = await api.delete(
      `/equipment/${institutionCode}/${labCode}/${equipmentCode}`
    );
    return response.data;
  },
};

export default equipmentService;